### ISSUE TITLE
Implement logical operators in searches (and, or, not, etc)

### ISSUE THREAD

**Issue by TallonRain** [Rothera/bpm#21]
Right now, BPM searches are limited to merely combined terms. For example, `+Fluttershy +luna` will return only emotes that include Fluttershy AND luna in them. This is the only search method possible outside of searching for specific subreddits. Using searches like `sr:mylittlepony sr:shymotes` return those subreddits, but are still AND searches.

A modifier to this such as `+fluttershy, +luna` or `+fluttershy || +luna` would then read as `fluttershy OR luna` and thus return all Fluttershy and all Luna emotes, like would would appear in exclusive `+fluttershy` or exclusive `+luna` searches.

To allow AND searches, either use a comma, ie `+fluttershy, +luna` to combine separate terms or just continue to use a simple space like what's used right now.

In short, implement logical operators in searches.

**Comment by TambourineReindeer** [comment]
Thank you! This would be nice to have, and I had been thinking of this myself.
